src/Microsoft.VisualStudio.Composition/MetadataViewClassDefaultCtorProvider.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.VisualStudio.Composition
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.Composition.Reflection;
+
+    /// <summary>
+    /// Supports metadata views that are concrete classes with a public default constructor
+    /// and properties with set accessors.
+    /// </summary>
+    internal class MetadataViewClassDefaultCtorProvider : IMetadataViewProvider
+    {
+        internal static readonly ComposablePartDefinition PartDefinition =
+            Utilities.GetMetadataViewProviderPartDefinition(typeof(MetadataViewClassDefaultCtorProvider), 1100000, Resolver.DefaultInstance);
+
+        internal static readonly IMetadataViewProvider Default = new MetadataViewClassDefaultCtorProvider();
+
+        private MetadataViewClassDefaultCtorProvider()
+        {
+        }
+
+        public bool IsMetadataViewSupported(Type metadataType)
+        {
+            Requires.NotNull(metadataType, nameof(metadataType));
+            var typeInfo = metadataType.GetTypeInfo();
+
+            return typeInfo.IsClass && !typeInfo.IsAbstract && FindConstructor(typeInfo) != null;
+        }
+
+        public object CreateProxy(IReadOnlyDictionary<string, object> metadata, IReadOnlyDictionary<string, object> defaultValues, Type metadataViewType)
+        {
+            Requires.NotNull(metadata, nameof(metadata));
```

Comment:
If we check that `metadata` is not null, should we do the same for `defaultValues`? Or, because as defined in the interface they can never be null, can we avoid the check altogether?

Reply:
I missed that. I'll add the `defaultValues` check. 
The interface xml doc is just that: documentation. It makes it 'legal' for an implementation to throw if the contract is violated. I generally code defensively like this because NullRefExceptions are just too painful to debug later. I only skip input/result verification when it's a concrete type that I have a _lot_ of trust in (like BCL types).